io/io/src/TStreamerInfoActions.cxx

```diff
@@ -226,6 +227,220 @@ namespace TStreamerInfoActions
       return 0;
    }
 
+   INLINE_TEMPLATE_ARGS Int_t WriteTextTNamed(TBuffer &buf, void *addr, const TConfiguration *config)
+   {
+      void *x = (void*)( ((char*)addr) + config->fOffset );
+      // Idea: Implement buf.ReadBasic/Primitive to avoid the return value
+      buf.StreamObject(x, TNamed::Class(), TNamed::Class());
+      return 0;
+   }
+
+   INLINE_TEMPLATE_ARGS Int_t WriteTextTObject(TBuffer &buf, void *addr, const TConfiguration *config)
+   {
+      void *x = (void*)( ((char*)addr) + config->fOffset );
+      // Idea: Implement buf.ReadBasic/Primitive to avoid the return value
+      buf.StreamObject(x, TObject::Class(), TObject::Class());
+      return 0;
+   }
+
+   /** Direct copy of code from TStreamerInfo::WriteBufferAux,
+    * potentially can be used later for non-text streaming */
+   template<bool isText>
+   INLINE_TEMPLATE_ARGS Int_t WriteSTLp(TBuffer &buf, void *addr, const TConfiguration *config)
+   {
+      TClass *cl                 = config->fCompInfo->fClass;
+      TMemberStreamer *pstreamer = config->fCompInfo->fStreamer;
+      TVirtualCollectionProxy *proxy = cl->GetCollectionProxy();
+      TClass* vClass = proxy ? proxy->GetValueClass() : 0;
+      UInt_t eoffset = 0; // extra parameter of TStreamerInfo::WriteBufferAux, 0 for all kind of objects writing
+      UInt_t ioffset = eoffset + config->fOffset;
+
+      if (!buf.TestBit(TBuffer::kCannotHandleMemberWiseStreaming)
+          && proxy && vClass
+          && config->fInfo->GetStreamMemberWise()
+          && cl->CanSplit()
+          && !(strspn(config->fCompInfo->fElem->GetTitle(),"||") == 2)
+          && !(vClass->TestBit(TClass::kHasCustomStreamerMember)) ) {
+         // Let's save the collection member-wise.
+
+         UInt_t pos = buf.WriteVersionMemberWise(config->fInfo->IsA(),kTRUE);
+         buf.WriteVersion( vClass, kFALSE );
+
+         // TODO: subinfo used for WriteBufferSTL call, which is private for the moment
+         //TStreamerInfo *subinfo = (TStreamerInfo*)vClass->GetStreamerInfo();
+
+         //for (int k = 0; k < narr; ++k) {
```

Comment:
> I was pushing you in that direction as it will increase the test coverage and benefit all users ;)

I prefer do it in small steps.
We have functional code and can incrementally improve it.
My main motivation now - clearly separate binary and text streaming from each other.
Once it is done, we could remove many workarounds from both.